lib/spack/spack/cmd/buildcache.py

```diff
@@ -59,6 +59,10 @@ def setup_parser(subparser):
     create.add_argument('-d', '--directory', metavar='directory',
                         type=str, default='.',
                         help="directory in which to save the tarballs.")
+    create.add_argument('-i', '--include_full_hash', action='store_true',
```

Comment:
I would just do this and the index by default.  Having them there should not hurt.

Reply:
Sounds good.